src/cata_tiles.h

```diff
@@ -126,6 +128,78 @@ struct tile_drawing_cache {
     }
 };
 
+struct pixel {
+    int r;
+    int g;
+    int b;
+    int a;
+
+    pixel()
+    {
+        r = 0;
+        g = 0;
+        b = 0;
+        a = 0;
+    }
+
+    pixel(SDL_Color c)
+    {
+        r = c.r;
+        g = c.g;
+        b = c.b;
+        a = c.a;
+    }
+
+    SDL_Color getSdlColor()
+    {
+        SDL_Color c;
+        c.r = static_cast<Uint8>(r);
+        c.g = static_cast<Uint8>(g);
+        c.b = static_cast<Uint8>(b);
+        c.a = static_cast<Uint8>(a);
+        return c;
+    }
+
+    bool isBlack()
+    {
+        return (r == 0 && g == 0 && b == 0);
+    }
+
+    operator==(const pixel &other)
+    {
+        return (r == other.r && g == other.g && b == other.b && a == other.a);
+    }
+    operator!=(const pixel &other)
+    {
+        return (r != other.r || g != other.g || b != other.b || a != other.a);
+    }
+};
+
+struct minimap_submap_cache {
+    //the color stored for each submap tile
+    std::vector< std::vector< pixel > > minimap_colors;
```

Comment:
It *might* be better to use a single vector and accessing its elements with `index = x * SEEY + y`. Currently this does 13 (SEEX + 1) heap allocation that can be all over the place, having the pixels in one chunk of memory (one vector) might be better.